### DISCUSSION TITLE
Public wishlist

### DISCUSSION THREAD

**Discussion by chris114782** [cmintey/wishlist#113] [Ideas]
I'd love to use this as I've been playing with it and it seems perfect for giving wish lists to friends and family for me. But if they have to sign up to view them then they essentially wont. It's too high a barrier for entry.

I'd like to be able to generate a public URL (maybe something like `https://wishlist.mydomain.com/share/<guid>`) that requires no authentication to view a list in a read only format.

**Comment by cmintey** [comment]
So is your thought that you would be the only user and the only person with a list, then you want to send out a public list so friends/family can view it, but not claim anything?

**Reply by chris114782** [reply]
Yeah. Exactly that.

If you compare it to say Amazon wishlists, they require no authentication to view a public list.

**Reply by cmintey** [reply]
If nobody can claim anything, how would they know another friend hasn't already bought an item for you?

I had a thought in my list of having a "registry mode" toggle which would do as you described, but also allow users to claim items, though there would need to be some identifier, not necessarily a sign-up, but an email or pin, to link claims to a user in case that same user wanted to un-claim

**Reply by chris114782** [reply]
I suppose claiming isn't really a feature I consider important, the old behavior of amazon wishlists (before they removed third party product links) didn't support this either, people just had to communicate with each other to say what they were getting.

I am looking to use something like this for making a registry of things I'd like and let people deal with it as is. All in public.

**Reply by cmintey** [reply]
I'll take a look at this soon when I have some time

**Reply by tpansino** [reply]
I'm also interested in this feature, and I see #116 is currently in Draft. Do you need help getting that finished up @cmintey ? I can help with testing out Docker builds in particular. Also coding, though I don't know Svelte.

[Wishthis](https://github.com/wishthis/wishthis) might also be worth looking at for an example implementation. That tool doesn't actually have the concept of claims, just the ability to "share" a list publicly via a link with a randomly generated slug, and the ability of anyone with that link to mark an item as "fulfilled", after which the item is deleted from the list (I think).

I think the idea of a placing a claim is actually much better, in particular the idea of being able to unclaim something, feels valuable.

Is there a point at which claimed items are deleted or hidden? Or are they retained indefinitely?

**Reply by cmintey** [reply]
I should be able to finish up that PR this week. The implantation I went with is similar to a registry. You have to set your list into registry mode, then you can add stuff to your list. You can get a public link to send out and users with that link will be able to claim items (they just need to provide a name, can be real name or a username, doesn't matter). Public users themselves cannot unclaim items since they don't have accounts.

> Is there a point at which claimed items are deleted or hidden? Or are they retained indefinitely?

Claims are retained indefinitely, though the owner of the list can remove items. There is also a button in the admin settings (either global or per group) that can clean up all claimed items.

**Reply by cmintey** [reply]
I just merged this feature. It should be available soon. Let me know if you run into any issues!

**Reply by tpansino** [reply]
Meant to comment on this thread sooner to say - this works great! Thanks!
